blockchain/chain/src/store/base_fee.rs

```diff
@@ -10,40 +10,36 @@ use fvm_ipld_encoding::Cbor;
 use fvm_shared::{clock::ChainEpoch, BLOCK_GAS_LIMIT};
 
 /// Used in calculating the base fee change.
-pub const BLOCK_GAS_TARGET: i64 = BLOCK_GAS_LIMIT / 2;
+pub const BLOCK_GAS_TARGET: u64 = BLOCK_GAS_LIMIT as u64 / 2;
 
 /// Limits gas base fee change to 12.5% of the change.
 pub const BASE_FEE_MAX_CHANGE_DENOM: i64 = 8;
 
 /// Genesis base fee.
 pub const INITIAL_BASE_FEE: i64 = 100000000;
-pub const PACKING_EFFICIENCY_DENOM: i64 = 5;
-pub const PACKING_EFFICIENCY_NUM: i64 = 4;
+pub const PACKING_EFFICIENCY_DENOM: u64 = 5;
+pub const PACKING_EFFICIENCY_NUM: u64 = 4;
 pub const MINIMUM_BASE_FEE: i64 = 100;
 
 fn compute_next_base_fee(
     base_fee: &TokenAmount,
-    gas_limit_used: i64,
+    gas_limit_used: u64,
     no_of_blocks: usize,
     epoch: ChainEpoch,
     smoke_height: ChainEpoch,
 ) -> TokenAmount {
-    let mut delta: i64 = if epoch > smoke_height {
-        (gas_limit_used / no_of_blocks as i64) - BLOCK_GAS_TARGET
+    let mut delta: u64 = if epoch > smoke_height {
+        (gas_limit_used / no_of_blocks as u64) - BLOCK_GAS_TARGET
```

Comment:
u64 subtraction looks dangerous. The original logic does not guarantee delta to be non-negative

Reply:
Reverted